### PR TITLE
Add AutomationProperties.Name to NumberBox

### LINKED ISSUE #325: Name Property missing for Progress text box present under header 'A determinate Progress bar'.
**Describe the bug**

**Steps to reproduce the bug**

1. Launch XAML Controls Gallery.
2. Activate ProgressBar from home page.
3. Navigate to text box under header A determinate progress bar

**Expected behavior**

Name Property should be there for text box,

**Actual behavior**

Text box name property missing.

**Screenshots**

image

**Version Info**

Gallery version (found on Settings page): 

1.2.12.0

**Additional context**

Copied from [internal bug](https://dev.azure.com/microsoft/OS/_workitems/edit/24649907)

### PR DESCRIPTION
**Pull request by karenbtlai**
## Description

Add AutomationProperties.Name to NumberBox to pass all Accessibility Insights test

## Motivation and Context

Fixes #325

### CODE REVIEW

File: XamlControlsGallery/ControlPages/ProgressBarPage.xaml [hunk 1]
```diff
@@ -47,9 +47,15 @@
                 <muxc:ProgressBar Width="130" x:Name="ProgressBar2"/>
                 <TextBlock x:Name="Control2Output" Style="{ThemeResource OutputTextBlockStyle}" Width="60" TextAlignment="Center" />
                 <TextBlock x:Name="ProgressLabel"  Text="Progress" VerticalAlignment="Center" Margin="0,0,10,0"/>
-                <muxc:NumberBox x:Name="ProgressValue" AutomationProperties.LabeledBy="{Binding ElementName=ProgressLabel}"
-                    Minimum="0" Maximum="100" Value="0"
-                    SpinButtonPlacementMode="Inline" ValueChanged="ProgressValue_ValueChanged"/>
+                <muxc:NumberBox
+                    x:Name="ProgressValue"
+                    AutomationProperties.LabeledBy="{Binding ElementName=ProgressLabel}"
+                    AutomationProperties.Name="NumberBox controlling ProgressBar2 value"
```
Comment (chingucoding) [inline XamlControlsGallery/ControlPages/ProgressBarPage.xaml:53]: Shouldn't setting the LabeledBy property be enough here? Could this actually be an issue with the NumberBox control?

Comment (karenbtlai) [reply XamlControlsGallery/ControlPages/ProgressBarPage.xaml:53]: LabeledBy applies the automationProperties.Name property to NumberBox but not the textbox, but mysteriously, the .Name property applies the name to both. This behaviour is duplicated with ComboBox. This is a temporary fix while I investigate the discrepancies between the two properties.

Comment (chingucoding) [reply XamlControlsGallery/ControlPages/ProgressBarPage.xaml:53]: I see, very strange that it works for one of the properties but not for the other. Could we template bind to the labeled by property maybe to mitigate this?

**Review by chingucoding — APPROVED** [review]
Fine to merge this as workaround until there is a proper fix on NumberBox. Thank you for fixing this!

File: XamlControlsGallery/ControlPages/ProgressBarPage.xaml (hunk 1 above)
Comment (karenbtlai) [reply XamlControlsGallery/ControlPages/ProgressBarPage.xaml:53]: One option is to templatebind InputBox to the header name like ComboBox, but that still doesn't solve the issue if there is no header available.

**Review by ranjeshj — APPROVED** [review]

**Pull request merged**